pagopa/io-app
# feat(Piattaforma Notifiche): [IAMVL-78] Show a disclaimer to the user before opening a message from PN

## Short description
### ⚠️ **This PR depends on #4010** 

This PR introduces a new feature (needed by PN) that alert the user before opening a message of a certain category. Then the user can choose to open the message or ignore it.

| Alert |
| - |
| <img src="https://user-images.githubusercontent.com/467098/177575614-bcd0463a-a7b4-4ed7-ab99-8b87840bf9ee.png" width="400" /> |

⚠️ **Beware that both the UI and the copy must be finalized and might change before releasing the feature.**

**Please note also the the "Don't show me again" logic will be added in #4036.**
 
## List of changes proposed in this pull request
- Added the new `useMessageOpening` hook that returns an `open` function (that will be called when the user taps on a message item) and a `bottomSheets` array (that will be added as child to the screen component);
- The hook will check for the message category and present a bottom sheet if needed (otherwise it will navigate to the message details).

## How to test
Use the [IIP-18-add-pn-endpoints](https://github.com/pagopa/io-dev-api-server/tree/IIP-18-add-pn-endpoints) branch of the dev server and add this to your `config.json`:
```
{
  "messages": {
    "pnCount": 1
  },
  "services": {
    "includePn": true
  }
}
```
Set `PN_ENABLED=YES` to `.env`, run the app and try to open a message from PN. Also make sure that opening another kind of message the navigation is dispatched.
